data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditableAuthorisedDataRequest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.model;
+
+
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * The class contains the authorised access to the resource.  This class is a container for the original data,
+ * {@link DataRequest}, the reference to the data that is authorised to be seen, {@link AuthorisedDataRequest} and
+ * {@link AuditErrorMessage} processing.  The container will hold the {@code AuditErrorMessage} if there was an error
+ * in getting the authorised data.
+ */
+public final class AuditableAuthorisedDataRequest {
+
+    private final DataRequest dataRequest;
+    private final AuthorisedDataRequest authorisedDataRequest;
+    private final AuditErrorMessage auditErrorMessage;
+
+    private AuditableAuthorisedDataRequest(
+            final DataRequest dataRequest,
+            final AuthorisedDataRequest authorisedDataRequest,
+            final AuditErrorMessage auditErrorMessage) {
+
+        this.dataRequest = dataRequest;
+        this.authorisedDataRequest = authorisedDataRequest;
+        this.auditErrorMessage = auditErrorMessage;
+    }
+
+    @Generated
+    public DataRequest getDataRequest() {
+        return dataRequest;
+    }
+
+    @Generated
+    public AuthorisedDataRequest getAuthorisedDataRequest() {
+        return authorisedDataRequest;
+    }
+
+
+    @Generated
+    public AuditErrorMessage getAuditErrorMessage() {
+        return auditErrorMessage;
```

Comment:
Return `Optional`.

Reply:
Will raise as ticked to cover this across all of the services